R/ResamplingSpCVBuffer.R

```diff
@@ -72,47 +89,55 @@ ResamplingSpCVBuffer = R6Class("ResamplingSpCVBuffer",
     #'   Returns the number of resampling iterations, depending on the
     #'   values stored in the `param_set`.
     iters = function() {
-      as.integer(length(self$instance$fold))
+      as.integer(length(self$instance))
     }
   ),
 
   private = list(
-    .sample = function(ids, coords, crs) {
+    .sample = function(ids, response, coords, positive, crs, properties) {
 
       require_namespaces(c("blockCV", "sf"))
 
-      points = sf::st_as_sf(coords,
+      pars = self$param_set$get_values()
+
+      if (!isTRUE("twoclass" %in% properties) && isTRUE(pars$spDataType == "PB")) {
+        stop("spDataType = 'PB' should only be used with two-class response")
+      }
+
+      if (!is.null(pars$addBG) && isTRUE(pars$spDataType == "PA")) {
+        stop("Parameter addBG should only be used with spDataType = 'PB'")
+      }
+
+      # Recode response to 0/1 for twoclass
+      if (isTRUE("twoclass" %in% properties)) {
```

Comment:
`isTRUE()` is problematic due to differing behavior in R > 3.5. Can be simply omitted here.